ovos_gui_plugin_shell_companion/brightness.py

```diff
@@ -168,51 +155,55 @@ def set_brightness_from_bus(self, message):
+        # dim screen in 60 seconds
+        self.event_scheduler.schedule_event(self.handle_dim_screen,
+                                            when=now_local() + timedelta(seconds=60), # TODO - seconds from config
+                                            name="ovos-shell.autodim")
+
+    def handle_dim_screen(self, message=None):
+        if self.auto_dim_enabled:
+            LOG.debug("Auto-dim: Lowering brightness")
+            self.bus.emit(Message("phal.brightness.control.auto.dim.update",
+                                  {"brightness": 20}))
+            self.set_brightness(20)  # TODO - value from enum
+
+    def _restore(self):
+        if self._brightness_level == 20:
+            LOG.debug("Auto-dim: Restoring brightness")
             if self.device_interface == "HDMI":
-                current_brightness = 100
+                self.set_brightness(100)  # TODO - value from enum
             if self.device_interface == "DSI":
-                current_brightness = 255
-
-            self.bus.emit(
-                Message("phal.brightness.control.auto.dim.update", {"brightness": 20}))
-            self.set_brightness(20)
+                self.set_brightness(255)  # TODO - value from enum
 
     def stop_auto_dim(self):
-        LOG.debug("Stopping Auto Dim")
-        self.auto_dim_enabled = False
-        if self.timer_thread:
-            self.timer_thread.join()
-
-    def restart_auto_dim(self):
-        LOG.debug("Restarting Auto Dim")
-        self.stop_auto_dim()
-        self.auto_dim_enabled = True
-        self.start_auto_dim()
-
-    def undim_display(self, message=None):
         if self.auto_dim_enabled:
-            LOG.debug("Undimming display on interaction")
-            if self.device_interface == "HDMI":
-                self.set_brightness(100)
-            if self.device_interface == "DSI":
-                self.set_brightness(255)
-            self.bus.emit(
-                Message("phal.brightness.control.auto.dim.update", {"brightness": "100"}))
-            self.restart_auto_dim()
-        else:
-            pass
-
-        ##### AUTO NIGHT MODE HANDLING #####
-
+            LOG.debug("Stopping Auto Dim")
+            self.config["auto_dim"] = False
+            update_config("auto_dim", False)
+            # cancel the next unfired dim event
+            self.event_scheduler.cancel_scheduled_event("ovos-shell.autodim")
+            self._restore()
+
+    def handle_undim_screen(self, message=None):
+        if self.auto_dim_enabled:
+            self._restore()
+            # schedule next auto-dim
+            self.event_scheduler.schedule_event(self.handle_dim_screen,
+                                                when=now_local() + timedelta(seconds=60),
+                                                name="ovos-shell.autodim")
+
+    ##################################
+    # AUTO NIGHT MODE HANDLING
+    # TODO - allow to do it based on camera, reacting live to brightness,
```

Comment:
I'd implement this as a separate feature; time-based and brightness-based adjustments are both valid use cases IMO

Reply:
agree, just wanted to leave the thought somewhere